From b612c1a61b4af226f8fbcb8e65f0438c8f106269 Mon Sep 17 00:00:00 2001
From: Andrew Purtell <apurtell@apache.org>
Date: Fri, 13 Feb 2015 14:48:49 -0800
Subject: [PATCH] Amend HBASE-9531 a command line (hbase shell) interface to
 retreive the replication metrics and show replication lag (Ashish Singhi)

Fix complilation for Hadoop 1 build
---
 .../regionserver/MetricsReplicationGlobalSourceSource.java         | 7 +++++++
 .../replication/regionserver/MetricsReplicationSinkSourceImpl.java | 7 +++++++
 .../regionserver/MetricsReplicationSourceSourceImpl.java           | 7 +++++++
 3 files changed, 21 insertions(+)

diff --git a/hbase-hadoop1-compat/src/main/java/org/apache/hadoop/hbase/replication/regionserver/MetricsReplicationGlobalSourceSource.java b/hbase-hadoop1-compat/src/main/java/org/apache/hadoop/hbase/replication/regionserver/MetricsReplicationGlobalSourceSource.java
index b16a751..8c47fa5 100644
--- a/hbase-hadoop1-compat/src/main/java/org/apache/hadoop/hbase/replication/regionserver/MetricsReplicationGlobalSourceSource.java
+++ b/hbase-hadoop1-compat/src/main/java/org/apache/hadoop/hbase/replication/regionserver/MetricsReplicationGlobalSourceSource.java
@@ -24,6 +24,7 @@ import org.apache.hadoop.metrics2.lib.MetricMutableGaugeLong;
 public class MetricsReplicationGlobalSourceSource implements MetricsReplicationSourceSource {
 
   private final MetricMutableGaugeLong ageOfLastShippedOpGauge;
+  private long ageOfLastShipped; // Hadoop 1 metrics don't let you read from gauges
   private final MetricMutableGaugeLong sizeOfLogQueueGauge;
   private final MetricMutableCounterLong logReadInEditsCounter;
   private final MetricMutableCounterLong logEditsFilteredCounter;
@@ -53,6 +54,7 @@ public class MetricsReplicationGlobalSourceSource implements MetricsReplicationS
 
   @Override public void setLastShippedAge(long age) {
     ageOfLastShippedOpGauge.set(age);
+    ageOfLastShipped = age;
   }
 
   @Override public void setSizeOfLogQueue(int size) {
@@ -93,4 +95,9 @@ public class MetricsReplicationGlobalSourceSource implements MetricsReplicationS
 
   @Override public void clear() {
   }
+
+  @Override
+  public long getLastShippedAge() {
+    return ageOfLastShipped;
+  }
 }
diff --git a/hbase-hadoop1-compat/src/main/java/org/apache/hadoop/hbase/replication/regionserver/MetricsReplicationSinkSourceImpl.java b/hbase-hadoop1-compat/src/main/java/org/apache/hadoop/hbase/replication/regionserver/MetricsReplicationSinkSourceImpl.java
index e69697b..a1609f5 100644
--- a/hbase-hadoop1-compat/src/main/java/org/apache/hadoop/hbase/replication/regionserver/MetricsReplicationSinkSourceImpl.java
+++ b/hbase-hadoop1-compat/src/main/java/org/apache/hadoop/hbase/replication/regionserver/MetricsReplicationSinkSourceImpl.java
@@ -24,6 +24,7 @@ import org.apache.hadoop.metrics2.lib.MetricMutableGaugeLong;
 public class MetricsReplicationSinkSourceImpl implements MetricsReplicationSinkSource {
 
   private final MetricMutableGaugeLong ageGauge;
+  private long ageOfLastApplied; // Hadoop 1 metrics don't let you read from gauges
   private final MetricMutableCounterLong batchesCounter;
   private final MetricMutableCounterLong opsCounter;
 
@@ -35,6 +36,7 @@ public class MetricsReplicationSinkSourceImpl implements MetricsReplicationSinkS
 
   @Override public void setLastAppliedOpAge(long age) {
     ageGauge.set(age);
+    ageOfLastApplied = age;
   }
 
   @Override public void incrAppliedBatches(long batches) {
@@ -44,4 +46,9 @@ public class MetricsReplicationSinkSourceImpl implements MetricsReplicationSinkS
   @Override public void incrAppliedOps(long batchsize) {
     opsCounter.incr(batchsize);
   }
+
+  @Override
+  public long getLastAppliedOpAge() {
+    return ageOfLastApplied;
+  }
 }
diff --git a/hbase-hadoop1-compat/src/main/java/org/apache/hadoop/hbase/replication/regionserver/MetricsReplicationSourceSourceImpl.java b/hbase-hadoop1-compat/src/main/java/org/apache/hadoop/hbase/replication/regionserver/MetricsReplicationSourceSourceImpl.java
index 0ea9ce0..b3acd6c 100644
--- a/hbase-hadoop1-compat/src/main/java/org/apache/hadoop/hbase/replication/regionserver/MetricsReplicationSourceSourceImpl.java
+++ b/hbase-hadoop1-compat/src/main/java/org/apache/hadoop/hbase/replication/regionserver/MetricsReplicationSourceSourceImpl.java
@@ -35,6 +35,7 @@ public class MetricsReplicationSourceSourceImpl implements MetricsReplicationSou
   private final String logReadInBytesKey;
 
   private final MetricMutableGaugeLong ageOfLastShippedOpGauge;
+  private long ageOfLastShipped; // Hadoop 1 metrics don't let you read from gauges
   private final MetricMutableGaugeLong sizeOfLogQueueGauge;
   private final MetricMutableCounterLong logReadInEditsCounter;
   private final MetricMutableCounterLong logEditsFilteredCounter;
@@ -74,6 +75,7 @@ public class MetricsReplicationSourceSourceImpl implements MetricsReplicationSou
 
   @Override public void setLastShippedAge(long age) {
     ageOfLastShippedOpGauge.set(age);
+    ageOfLastShipped = age;
   }
 
   @Override public void setSizeOfLogQueue(int size) {
@@ -126,4 +128,9 @@ public class MetricsReplicationSourceSourceImpl implements MetricsReplicationSou
 
     rms.removeMetric(logEditsFilteredKey);
   }
+
+  @Override
+  public long getLastShippedAge() {
+    return ageOfLastShipped;
+  }
 }
-- 
1.9.1

